### PR TITLE
Dev dotnet day 2

### PR DESCRIPTION
**Pull request by Sizuni**

### CODE REVIEW

File: DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsByShapeService.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        {
+            ResultDictionary.Add("A X", 4); // Rock Rock
+            ResultDictionary.Add("A Y", 8); // Rock Paper
+            ResultDictionary.Add("A Z", 3); // Rock Scissors
+            ResultDictionary.Add("B X", 1); // Paper Rock
+            ResultDictionary.Add("B Y", 5); // Paper Paper
+            ResultDictionary.Add("B Z", 9); // Paper Scissors
+            ResultDictionary.Add("C X", 7); // Scissors Rock
+            ResultDictionary.Add("C Y", 2); // Scissors Paper
+            ResultDictionary.Add("C Z", 6); // Scissors Scissors
```
Comment (Fblade25) [inline DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsByShapeService.cs:15]: The C# way of predefining answers

File: DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsService.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        {
+            ResultDictionary = new Dictionary<string, int>();
+        }
+
+        public int GetRoundResult(string input) {
+            if (ResultDictionary.TryGetValue(input, out int result))
+            {
+                return result;
+            }
+            throw new KeyNotFoundException($"No value found for key: {input}");
```
Comment (Fblade25) [inline DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsService.cs:19]: imagine error handling

**Review by Fblade25 — APPROVED** [review]
dictionaries strong

File: DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsService.cs (hunk 2 above)
Comment (Sizuni) [reply DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsService.cs:19]: yes

File: DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsByShapeService.cs (hunk 1 above)
Comment (Sizuni) [reply DotNet/AdventOfCode2022/Day02RockPaperScissors/Services/RockPaperScissorsByShapeService.cs:15]: because the score for every combination is fixed

**Pull request merged**